master/docs/manual/configuration/reporters.rst

```diff
@@ -785,6 +787,236 @@ Two additional arguments can be set to control how fast the IRC bot tries to rec
 ``lostDelay`` defaults to a random number between 1 and 5, while ``failedDelay`` defaults to a random one between 45 and 60.
 Setting random defaults like this means multiple IRC bots are less likely to deny each other by flooding the server.
 
+.. bb:reporter:: TelegramBot
+
+Telegram Bot
+~~~~~~~~~~~~
+
+Buildbot offers a bot, similar to the :bb:reporter:`IRC` for Telegram mobile and desktop messaging app. The bot can notify users and groups about build events, respond to status queries, or force and stop builds on request (if allowed to).
+
+In order to use this reporter, you must first speak to BotFather_ and create a `new telegram bot <https://core.telegram.org/bots#creating-a-new-bot>`. A quick step-by-step procedure is as follows:
+
+1. Start a chat with BotFather_.
+
+2. Type ``/newbot``.
+
+3. Enter a display name for your bot. It can be any string.
+
+4. Enter a unique username for your bot. Usernames are 5-32 characters long and are case insensitive, but may only include Latin characters, numbers, and underscores. Your bot's username must end in `bot`, e.g. `MyBuildBot` or `MyBuildbotBot`.
+
+5. You will be presented with a token for your bot. Save it, as you will need it for :bb:reporter:`TelegramBot` configuration.
+
+6. Optionally, you may type ``/setcommands``, select the username of your new bot and paste the following text:
```

Comment:
Couldn't this be done more automatically?

I would prefer this list to be only generated by the bot rather than hardcoded in the doc.

Reply:
Can we add the result of some code to docs? Right now, the bot command `/commands botfather` can print the list to copy-paste to Bot Father. Telegram provides no way to do it automatically by bot.

Reply:
We are already generating some of the doc:

https://github.com/buildbot/buildbot/blob/cc304b5a4e656200853a41ba8abc6e3b3a58cb57/master/docs/developer/rest.rst#L188-L200

https://pypi.org/project/sphinx-jinja/

https://github.com/buildbot/buildbot/blob/cc304b5a4e656200853a41ba8abc6e3b3a58cb57/master/docs/conf.py#L323-L326

Reply:
I didn't know, you can use Jinja in Sphinx. That's neat.